R/guide-axis-theta.R

```diff
@@ -0,0 +1,332 @@
+        top    = 0,
+        bottom = 1   * pi,
+        left   = 1.5 * pi,
+        right  = 0.5 * pi
+      )
+    } else {
+      if (params$position == 'theta.sec') {
+        key$theta <- key$theta + pi
+      }
+
+      # If the first and last positions are close together, we merge the
+      # labels of these positions
+      ends_apart <- (key$theta[n] - key$theta[1]) %% (2 * pi)
+      if (n > 0 && ends_apart < 0.05 && !is.null(key$.label)) {
+        if (is.expression(key$.label)) {
+          combined <- substitute(
+            paste(a, "/", b),
+            list(a = key$.label[[1]], b = key$.label[[n]])
+          )
+        } else {
+          combined <- paste(key$.label[1], key$.label[n], sep = "/")
+        }
+        key$.label[[n]] <- combined
+        key <- vec_slice(key, -1)
+      }
+    }
+
+    params$key <- key
+    params
+  },
+
+  setup_params = function(params) {
+    # Theta axis doesn't need to setup any position specific parameters.
+    params
+  },
+
+  setup_elements = function(params, elements, theme) {
+    axis_elem <- c("line", "text", "ticks", "ticks_length")
+    is_char <- vapply(elements[axis_elem], is.character, logical(1))
+    axis_elem <- axis_elem[is_char]
+
+    aes <- switch(
+      params$position,
+      theta     = "x.bottom",
+      theta.sec = "x.top",
+      paste0(params$aesthetic, ".", params$position)
+    )
+
+    elements[axis_elem] <- lapply(
+      paste(unlist(elements[axis_elem]), aes, sep = "."),
+      calc_element, theme = theme
+    )
+    elements$minor_ticks <- combine_elements(params$minor.ticks, elements$ticks)
+
+    # Offset distance from axis arc to text positions
+    offset <- max(0, params$major.length, params$minor.length)
+    offset <- max(elements$ticks_length * offset, unit(0, "pt")) +
+      max(elements$text$margin)
+    elements$offset <- offset
+    elements
+  },
+
+  override_elements = function(params, elements, theme) {
+    # We don't override any label angles/hjust/vjust because these depend on
+    # theta of label.
+    return(elements)
+  },
+
+  build_labels = function(key, elements, params) {
+
+    key <- vec_slice(key, !vec_detect_missing(key$.label %||% NA))
+
+    # Early exit if drawing no labels
+    labels <- key$.label
+    if (length(labels) < 1) {
+      return(list(zeroGrob()))
+    }
+
+    # Resolve text angle
+    if (is.waive(params$angle) || is.null(params$angle)) {
+      angle <- elements$text$angle
+    } else {
+      angle <- flip_text_angle(params$angle - rad2deg(key$theta))
+    }
+    # Text angle in radians
+    rad <- deg2rad(angle)
+    # Position angle in radians
+    theta <- key$theta
+
+    # Offset distance to displace text away from outer circle line
+    xoffset <- elements$offset * sin(theta)
+    yoffset <- elements$offset * cos(theta)
+
+    # Note that element_grob expects 1 angle for *all* labels, so we're
+    # rendering one grob per label to propagate angle properly
+    do.call(grobTree, Map(
```

Comment:
We have moved away from `do.call()` in favour of `rlang::inject()` when possible